### PR TITLE
1931 cda validity fixes

### PR DESCRIPTION
**Pull request by RamilGaripov**
Ticket: https://github.com/metriport/metriport-internal/issues/1931

### Description
- Adding a ton of fixes to the FHIR to CDA converter based on the C-CDA validator mentioned in the ticket. These sections now look really good:
  - Social History
  - Medications
  - Problems
  - Allergies
  - Encounters
  - Immunizations
  - Vital Signs
  - Family History
- ⚠️ Temporarily disabling the unit tests since the mappings are going to be changing rapidly for the next while

### Testing
- Local
  - [x] Test CDA with local validator
  - [x] Test CDA with online validator: 
    - Still not ideal, but much better than before

### Release Plan
- [ ] Merge this

### CODE REVIEW

File: packages/core/src/fhir-to-cda/cda-types/shared-types.ts [hunk 1]
```diff
@@ -51,7 +52,7 @@ export type CdaAddressUse =
   | "TMP"
   | "WP";
 
-export type CdaGender = "M" | "F" | "UK";
+export type CdaGender = "M" | "F" | "UN";
```
Comment (jonahkaye) [inline packages/core/src/fhir-to-cda/cda-types/shared-types.ts:55]: so ala all the gender convos we have had, UN = other and UNK = Unknown. I am pretty sure this is the case in CDAs too and not just IHE. Shouldnt this be UNK? Or did you get feedback from the online validator that UNK was invalid?

Comment (RamilGaripov) [reply packages/core/src/fhir-to-cda/cda-types/shared-types.ts:55]: Yes, the online validator was not ok with `UK`, but is good with `UN`. And on the gender mapping function, I posted the link to the CDA documentation that provides acceptable genders. It's listed as `UN` there - [link](https://terminology.hl7.org/5.2.0/ValueSet-v3-AdministrativeGender.html)

Comment (jonahkaye) [reply packages/core/src/fhir-to-cda/cda-types/shared-types.ts:55]: Ok but then what do we do for patients with gender UNK. We have those in our system now and we need to support them. Can we just not give a gender in those cases or will that throw issues on the validator too?

Comment (jonahkaye) [reply packages/core/src/fhir-to-cda/cda-types/shared-types.ts:55]: Technically in our system they are U but same point

**Review by jonahkaye — APPROVED** [review]
Approved, but just confirm that a null flavor works with the CDA validator tool youve been using if you havent already

File: packages/core/src/fhir-to-cda/cda-types/shared-types.ts (hunk 1 above)
Comment (RamilGaripov) [reply packages/core/src/fhir-to-cda/cda-types/shared-types.ts]: Discussed in slack, and fixed the issue.

**Pull request merged**